src/main/haskell/language-kore/app/parser/GlobalMain.hs

```diff
@@ -0,0 +1,161 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE RecordWildCards #-}
```

Comment:
Is using RecordWildCards really a good idea? Our current style guide does not say anything about it, but I may change it based on this discussion.

There are two problems that I see.

First, that it makes it harder to understand what happens there (i.e. that you have a projection which is implicitly applied to a record as opposed to a constant function), it requires a lookup for the projection name when reading the code instead of just glancing over the code.

Second, that it can overlap with a similarly named local thing, and it's not always trivial to say which is which.

Now, there is a good chance that --pedantic catches the second issue, but I didn't check.

About the first issue, the current style guide allows two ways of naming record fields: either with `RecordWildCards` and free-syle-ish names, but then one is not allowed to use them as getters/projections, or the fields should be prefixed with the record name.

Now, that almost solves the first problem: using projections is only allowed when they include the structure name, and the structure name is available locally, since one must match on it in order to have the implicit projection argument stuff.

Still, I can't help but think that code is more readable when using explicit arguments. What do you think?

Reply:
I was trying to think of the best way to introduce the options into the scope of the main function that would be the easiest to change, as I anticipate more flags and functionality being added.  But I will switch it back to the {NamedFieldPuns}  This would only mean adding another match to the main function when you add another option (compared to no change at all) but i agree it would be easier to read